sqlglot/parser.py

```diff
@@ -1709,6 +1709,13 @@ def _parse_function(self):
         return self._parse_window(this)
 
     def _parse_lambda(self):
+        def replace_lambda_columns_with_identifiers(node, lambda_variables):
```

Comment:
can you make this a first class function on the last line, after _replace_columns_with_dots

change lambda variables to be a set 

self._parse_conjunction().transform(self._replace_lambda, {node.name for node in expressions}